File: 1318.py
class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        ba=format(a, '032b')
        bb=format(b, '032b')
        bc=format(c, '032b')
        cnt=0
        for i,j,k in zip(ba,bb,bc):
            if k=='0':
                if i=='1':
                    cnt+=1
                if j=='1':
                    cnt+=1
            elif k=='1':
                if i=='1' or j=='1':
                    continue
                else:
                    cnt+=1    

        return cnt
        

File: 1318.cpp
class Solution {
public:
    int minFlips(int a, int b, int c) {
        int diff=(a|b)^c,cnt;
        int rep=a&b&diff;
        while(diff!=0 || rep!=0){
            cnt+=(1&diff)+(1&rep);
            diff=diff>>1;
            rep=rep>>1;
        }
        return cnt;
    }
};